config/routes.rb

```diff
@@ -2,5 +2,9 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  root "users#index"
+  get "/users/:user_id/posts" => "posts#index", as: :user_posts
+  get "/users/:user_id/posts/:id" => "posts#show", as: :user_post
+  get "/users" => "users#index", as: :users
+  get "/users/:id" => "users#show", as: :user
```

Comment:
- [ ] I think we can write more optimal code here. We don't need to define these routes like this. Rails already has helper called resources which we can use for this and expose only the routes we need with the only keyword, like so:
```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
It looks clean! Will do 💪